modules/javafx.graphics/src/main/native-prism-d3d/D3DPhongMaterial.h

```diff
@@ -30,10 +30,15 @@
 
 // See MaterialPhong.h, MaterialPhongShaders.h
 
-#define DIFFUSE 0
-#define SPECULAR 1
-#define BUMP 2
-#define SELFILLUMINATION 3
+// Map type numbered sequence used for sampling registers (we have 4 sampling registers for vs 3.0).
+// Order is defined by com.sun.prism.PhongMaterial's MapType enum
+enum map_type : unsigned int {
+    diffuse,
+    specular,
+    bump,
+    self_illumination,
+    num_map_types
+};
```

Comment:
About 2, I tried that initially and got into problems iterating over the enum in `D3DMeshView.cc` line 293. How would you perform the iteration?

Reply:
You're right, there's no implicit conversion from scoped enums to integers. While one could get very creative with C++, maybe just using an unscoped enum is easier. You might want to prefix the constants with `MT_` or something like that.

Reply:
The `BUMP` symbol is not scoped to `MapType`; prefixing with `MapType::` is neither required, nor changes its meaning. The symbol exists in the global namespace in all compilation units that include the header. Adding a prefix may help to avoid name collisions in the future.